hyper-boring/src/v1.rs

```diff
@@ -37,24 +42,38 @@ impl HttpsConnector<HttpConnector> {
     /// HTTP/2 and HTTP/1.1.
     ///
     /// Requires the `runtime` Cargo feature.
-    pub fn new() -> Result<HttpsConnector<HttpConnector>, ErrorStack> {
+    // This is a bit sad, but avoids requiring `TokioIo` as a response type below.
+    #[allow(clippy::type_complexity)]
+    pub fn new() -> Result<
+        HttpsConnector<
+            MapResponse<
+                HttpConnector,
+                impl FnOnce(TokioIo<TcpStream>) -> TokioIo<TokioIo<TcpStream>> + Clone,
```

Comment:
This could be `fn(TokioIo<TcpStream>)`, and then you'll be able to shove that type behind a `type` alias.

Reply:
Ah yes, good point, that's what I was looking for!

Reply:
Easily done. Cleaner indeed, thanks!